flashcards/web/src/components/AddDecksForm.tsx

```diff
@@ -0,0 +1,154 @@
+import React from "react";
+import { useFormik } from "formik";
+import {
+    Button,
+    Checkbox,
+    Container,
+    FormControlLabel,
+    Grid,
+    InputLabel,
+    MenuItem,
+    Select,
+    TextField,
+    Typography,
+} from "@mui/material";
+import * as Yup from "yup";
+
+function AddDecksForm() {
+    const formik = useFormik({
+        validateOnChange: false,
+        initialValues: {
+            tagName: "",
+            author: "",
+            selectTag: "",
+            selectCategory: "",
+            selectAvailability: "",
+            price: "",
+            checkboxPublic: false,
+            checkboxPrice: false,
+        },
+        onSubmit: (values) => {
+            alert(JSON.stringify(values, null, 2));
+        },
+        validationSchema: Yup.object().shape({
+            tagName: Yup.string().min(2, "Too Short!").max(20, "Too Long!").required("Required!"),
+            author: Yup.string().min(2, "Too Short!").max(20, "Too Long!").required("Required!"),
+            checkboxPrice: Yup.boolean(),
+            checkboxPublic: Yup.boolean(),
+            price: Yup.number().when("checkboxPrice", {
+                is: true,
+                then: Yup.number().required("Required!"),
+            }),
+            selectTag: Yup.string().required("Required!"),
+            selectCategory: Yup.string().required("Required!"),
+        }),
+    });
+    return (
+        <form onSubmit={formik.handleSubmit}>
+            <Container maxWidth="md">
+                <Grid container>
```

Comment:
I'm quite confused, you've used grid and set each element to width of 12.
The same behavior will be obtained without using any grid isn't it?

Reply:
Still not geting why you're using grid